messaging/messaging.go

```diff
@@ -214,36 +214,52 @@ func (p *APNSPayload) MarshalJSON() ([]byte, error) {
 // Alert may be specified as a string (via the AlertString field), or as a struct (via the Alert
 // field).
 type Aps struct {
-	AlertString      string    `json:"-"`
-	Alert            *ApsAlert `json:"-"`
-	Badge            *int      `json:"badge,omitempty"`
-	Sound            string    `json:"sound,omitempty"`
-	ContentAvailable bool      `json:"-"`
-	Category         string    `json:"category,omitempty"`
-	ThreadID         string    `json:"thread-id,omitempty"`
+	AlertString      string
+	Alert            *ApsAlert
+	Badge            *int
+	Sound            string
+	ContentAvailable bool
+	MutableContent   bool
+	Category         string
+	ThreadID         string
+	CustomData       map[string]interface{}
 }
 
-// MarshalJSON marshals an Aps into JSON (for internal use only).
-func (a *Aps) MarshalJSON() ([]byte, error) {
-	type apsAlias Aps
-	s := &struct {
-		Alert            interface{} `json:"alert,omitempty"`
-		ContentAvailable *int        `json:"content-available,omitempty"`
-		*apsAlias
-	}{
-		apsAlias: (*apsAlias)(a),
-	}
-
+func (a *Aps) standardFields() map[string]interface{} {
```

Comment:
Would be nice to add a short comment for this function and mention that it doesn't include CustomData.

Reply:
Done